homeassistant/components/monoprice/media_player.py

```diff
@@ -127,7 +131,12 @@ def __init__(self, monoprice, sources, namespace, zone_id):
 
     def update(self):
         """Retrieve latest state."""
-        state = self._monoprice.zone_status(self._zone_id)
+        try:
+            state = self._monoprice.zone_status(self._zone_id)
+        except SerialException:
+            _LOGGER.warning("Could not update zone %d", self._zone_id)
+            return False
```

Comment:
I'm not too happy with this - ideally I would have raised an `UpdateFailed` here. However, since this is not actually an error for Zones that aren't connected, there are two negative side effects to raising here:
1. The logs contain an error
2. The entities are never created (instead of created and disabled)

Reply:
Nothing is check the return value of `update`. Please remove those and just `return` where needed.